turbinia/state_manager.py

```diff
@@ -328,3 +332,146 @@ def write_new_task(self, task):
       log.error(f'Unsuccessful in writing new task {task.name:s} into Redis')
     task.state_key = key
     return key
+
+  def write_evidence(self, evidence_dict: dict[str]):
+    """Writes evidence into redis.
+
+    Args:
+      evidence_dict (dict[str]): A dictionary containing the serialized
+      evidence attributes that will be saved.
+
+    Returns:
+      key (str): The key corresponding to the evidence in redis
+    """
+    key = ':'.join(('TurbiniaEvidence', json.loads(evidence_dict['id'])))
+    log.info(f'Writing new evidence {key} into Redis')
+    for attribute_key, attribute_value in evidence_dict.items():
+      if not self.client.hset(key, attribute_key, attribute_value):
+        log.error(
+            f'Unsuccessful in writing {attribute_key} of {key} into Redis')
+    if evidence_hash := json.loads(evidence_dict.get('hash')):
+      self.client.hset('TurbiniaEvidenceHashes', evidence_hash, key)
+    return key
+
+  def update_evidence_attribute(
+      self, evidence_id: str, attribute_name: str, json_value: str):
+    """Updates one attribute of the evidence in Redis.
+
+    Args:
+      evidence_id (str): The ID of the stored evidence.
+      attribute_name (str): name of the attribute to be updated.
+      json_value (str): json value to be updated.
+    """
+    key = ':'.join(('TurbiniaEvidence', evidence_id))
+    log.info(f'Updating attribute {attribute_name} for evidence {key} in Redis')
+    if self.client.hset(key, attribute_name, json_value):
+      if attribute_name == 'hash' and json_value:
+        self.client.hset('TurbiniaEvidenceHashes', json.loads(json_value), key)
+
+  def get_evidence(self, evidence_id: str):
+    """Gets one evidence from Redis given its ID.
```

Comment:
I tried to keep everything using dictionaries instead of evidence in the state_manager to make it more standardized and to avoid circular imports. I renamed it here now to get_evidence_data